### PR TITLE
pass python version info to inner build

### PR DESCRIPTION
**Pull request by rdeits**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -63,13 +63,16 @@ endif()
 ExternalProject_Add(iris_project
 	SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src
     CMAKE_ARGS
+      -DBUILD_TESTING=${BUILD_TESTING}
       -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
       -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
-      -DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX}
       -DCMAKE_MODULE_PATH=${CMAKE_MODULE_PATH}
-      -DBUILD_TESTING=${BUILD_TESTING}
+      -DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX}
```
Comment (jamiesnape) [inline CMakeLists.txt]: Do you mean `-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}`?

Comment (rdeits) [inline CMakeLists.txt]: I have no idea what I meant there. I think that line can just be deleted.

**Comment by jamiesnape** [comment]
You will need to pass along the `CMAKE_PREFIX_PATH` and you probably need to prepend `CMAKE_INSTALL_PREFIX` to it. Do something like
```
list(INSERT CMAKE_PREFIX_PATH 0 "${CMAKE_INSTALL_PREFIX}")
...
-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -60,16 +60,27 @@ if (IRIS_WITH_PYBIND11)
...
-      -DBUILD_TESTING=${BUILD_TESTING}
       -DWITH_MATLAB=${IRIS_WITH_MATLAB}
-      -DPYBIND11_PYTHON_VERSION=${PYBIND11_PYTHON_VERSION}
+      -DPYBIND11_PYTHON_VERSION=${PYBIND11_PYTHON_VERSION})
+
+if (DEFINED PYTHON_EXECUTABLE)
+  set(IRIS_CMAKE_ARGS ${IRIS_CMAKE_ARGS} -DPYTHON_EXECUTABLE=${PYTHON_EXECUTABLE})
+endif()
+if (DEFINED PYTHON_INCLUDE_DIR)
+  set(IRIS_CMAKE_ARGS ${IRIS_CMAKE_ARGS} -DPYTHON_EXECUTABLE=${PYTHON_INCLUDE_DIR})
```
Comment (jamiesnape) [inline CMakeLists.txt]: This and below explain the failures in Drake.
```-D PYTHON_INCLUDE_DIR E=${PYTHON_INCLUDE_DIR}```

File: CMakeLists.txt [hunk 3]
```diff
@@ -60,16 +60,27 @@ if (IRIS_WITH_PYBIND11)
...
+endif()
+if (DEFINED PYTHON_LIBRARY)
+  set(IRIS_CMAKE_ARGS ${IRIS_CMAKE_ARGS} -DPYTHON_EXECUTABLE=${PYTHON_LIBRARY})
```
Comment (jamiesnape) [inline CMakeLists.txt]: ```-DPYTHON_LIBRARY=${PYTHON_LIBRARY}```

File: CMakeLists.txt (hunk 2 above)
Comment (rdeits) [inline CMakeLists.txt]: Wow, thank you. This is what I get for editing cmake files at 2am.

Comment (rdeits) [inline CMakeLists.txt]: Should be fixed in #54

File: CMakeLists.txt (hunk 3 above)
Comment (rdeits) [inline CMakeLists.txt]: fixed in #54

**Pull request merged**